lib/floe/workflow/intrinsic_function.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    class IntrinsicFunction
+      INTRINSIC_FUNCTION_REGEX = /^(?<module>\w+)\.(?<function>\w+)\((?<args>.*)\)$/.freeze
+
+      class << self
+        def klass(payload)
```

Comment:
Minor, but I think it's clearer to describe the name with something like

```suggestion
        def detect_class(payload)
```